uiowa/uiowa
# SPPA Private - Follow-up fixes

Resolves https://github.com/uiowa/uiowa/issues/4008
Relates to https://github.com/uiowa/uiowa/issues/3624

- Removes the [private field from node render](https://github.com/uiowa/uiowa/issues/4008#issuecomment-902255285).
- Changes pharmacy person teaser and palliative care map listing to support both "credentials" appending to name and territory if provided.
- Discard identical role.viewer config from sppa.

<img width="642" alt="Screen Shot 2021-08-20 at 4 23 40 PM" src="https://user-images.githubusercontent.com/4663676/130294749-eaf6c3f1-d2b6-4016-8739-5326ee9633b4.png">


There are some weird config issues going on with this split, but @pyrello said to ignore until we have a better idea of any changes in core and until we update config_split.

<details>
<summary>config split diff to ignore</summary>

```
Untracked files:
  (use "git add <file>..." to include in what will be committed)
	config/sppa.uiowa.edu/core.base_field_override.node.article.promote.yml
	config/sppa.uiowa.edu/core.base_field_override.node.page.promote.yml
	config/sppa.uiowa.edu/core.base_field_override.node.person.promote.yml
	config/sppa.uiowa.edu/core.entity_form_display.block_content.featured_content.default.yml
	config/sppa.uiowa.edu/core.entity_form_display.node.article.default.yml
	config/sppa.uiowa.edu/core.entity_form_display.node.page.default.yml
	config/sppa.uiowa.edu/core.entity_form_display.node.person.default.yml
	config/sppa.uiowa.edu/core.entity_form_display.node.person.minimal.yml
	config/sppa.uiowa.edu/entity_browser.browser.featured_content_browser.yml
	config/sppa.uiowa.edu/field.field.block_content.featured_content.field_featured_content_item.yml
	config/sppa.uiowa.edu/field.field.node.article.body.yml
	config/sppa.uiowa.edu/field.field.node.article.field_article_author.yml
	config/sppa.uiowa.edu/field.field.node.article.field_article_source_link.yml
	config/sppa.uiowa.edu/field.field.node.article.field_article_source_link_direct.yml
	config/sppa.uiowa.edu/field.field.node.article.field_article_source_org.yml
	config/sppa.uiowa.edu/field.field.node.article.field_featured_image_display.yml
	config/sppa.uiowa.edu/field.field.node.article.field_image.yml
	config/sppa.uiowa.edu/field.field.node.article.field_image_caption.yml
	config/sppa.uiowa.edu/field.field.node.article.field_meta_tags.yml
	config/sppa.uiowa.edu/field.field.node.article.field_tags.yml
	config/sppa.uiowa.edu/field.field.node.article.field_teaser.yml
	config/sppa.uiowa.edu/field.field.node.page.body.yml
	config/sppa.uiowa.edu/field.field.node.page.field_featured_image_display.yml
	config/sppa.uiowa.edu/field.field.node.page.field_image.yml
	config/sppa.uiowa.edu/field.field.node.page.field_meta_tags.yml
	config/sppa.uiowa.edu/field.field.node.page.field_publish_options.yml
	config/sppa.uiowa.edu/field.field.node.page.field_tags.yml
	config/sppa.uiowa.edu/field.field.node.page.field_teaser.yml
	config/sppa.uiowa.edu/field.field.node.page.layout_builder__layout.yml
	config/sppa.uiowa.edu/field.field.node.person.field_image.yml
	config/sppa.uiowa.edu/field.field.node.person.field_meta_tags.yml
	config/sppa.uiowa.edu/field.field.node.person.field_person_bio.yml
	config/sppa.uiowa.edu/field.field.node.person.field_person_contact_information.yml
	config/sppa.uiowa.edu/field.field.node.person.field_person_credential.yml
	config/sppa.uiowa.edu/field.field.node.person.field_person_education.yml
	config/sppa.uiowa.edu/field.field.node.person.field_person_email.yml
	config/sppa.uiowa.edu/field.field.node.person.field_person_first_name.yml
	config/sppa.uiowa.edu/field.field.node.person.field_person_hide.yml
	config/sppa.uiowa.edu/field.field.node.person.field_person_hometown.yml
	config/sppa.uiowa.edu/field.field.node.person.field_person_last_name.yml
	config/sppa.uiowa.edu/field.field.node.person.field_person_phone.yml
	config/sppa.uiowa.edu/field.field.node.person.field_person_position.yml
	config/sppa.uiowa.edu/field.field.node.person.field_person_research_areas.yml
	config/sppa.uiowa.edu/field.field.node.person.field_person_type_status.yml
	config/sppa.uiowa.edu/field.field.node.person.field_person_types.yml
	config/sppa.uiowa.edu/field.field.node.person.field_person_website.yml
	config/sppa.uiowa.edu/field.field.node.person.field_tags.yml
	config/sppa.uiowa.edu/field.field.node.person.field_teaser.yml
	config/sppa.uiowa.edu/views.view.articles.yml
	config/sppa.uiowa.edu/views.view.articles_block.yml
	config/sppa.uiowa.edu/views.view.featured_content.yml
	config/sppa.uiowa.edu/views.view.moderated_content.yml
	config/sppa.uiowa.edu/views.view.people.yml
	config/sppa.uiowa.edu/views.view.people_block.yml
	config/sppa.uiowa.edu/views.view.people_list_block.yml
	config/sppa.uiowa.edu/views.view.person_card.yml
	config/sppa.uiowa.edu/workflows.workflow.editorial.yml
```
</details>

## Stories

- As an editor of sppa.uiowa.edu I can mark a page as private or public, save and not have the field render on the node.
- As an editor of pharmacy.uiowa.edu I can create a Palliative care person with a location, credentials and have that information displayed on the Palliative care listing view block.

## To Test

- `blt ds --site=sppa.uiowa.edu && blt ds --site=pharmacy.uiowa.edu`
- Login into both sites
- For sppa, edit an existing page and mark it as private (promotion options). This should make the page access denied to anon viewers but should not render the private field to authenticated.
- For pharmacy, create a person with a territory and palliative care person type checked. Also include some credentials. When you place the listing view on a layout builder page, the person should show with credentials appended to the person name and the territory displayed as well.
